src/core/ext/xds/xds_bootstrap_grpc.cc

```diff
@@ -376,24 +367,36 @@ grpc_error_handle GrpcXdsBootstrap::ParseCertificateProvider(
         "\"plugin_name\" field is not a string"));
   } else {
     std::string plugin_name = std::move(*(it->second.mutable_string_value()));
+    // Find config JSON.
+    absl::optional<Json> config_json;
     it = certificate_provider_json->mutable_object()->find("config");
     if (it != certificate_provider_json->mutable_object()->end()) {
       if (it->second.type() != Json::Type::OBJECT) {
         error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
             "\"config\" field is not an object"));
       } else {
-        absl::Status status = certificate_provider_plugin_map_->AddPlugin(
-            instance_name, plugin_name, it->second);
-        if (!status.ok()) {
-          error_list.push_back(absl_status_to_grpc_error(status));
-        }
+        config_json = it->second;
       }
     } else {
-      // "config" is an optional field, so create an empty JSON object.
-      absl::Status status = certificate_provider_plugin_map_->AddPlugin(
-          instance_name, plugin_name, Json::Object());
-      if (!status.ok()) {
-        error_list.push_back(absl_status_to_grpc_error(status));
+      // "config" is an optional field, so use an empty JSON object.
```

Comment:
maybe a better comment would be - "config" is an optional field that defaults to an empty JSON object.